### ISSUE TITLE
Should keybindings be disabled by default?

### ISSUE THREAD

**Issue by BjarniRunar** [mailpile/Mailpile#1985]
Keybindings are a valuable feature, a necessity for power-users.

However, there is a case to be made for disabling them by default:

   1. Many users are unaware of keyboard short-cuts, and many users never use them
   2. Keyboard short-cuts can cause surprises, even data-loss: if you accidentally press a button, or your cat does, or your child does. Stuff happens and *you may not notice*.

In most cases, accidentally hitting buttons will trigger actions which the user should notice. This will be annoying (and make the user distrust Mailpile because they won't understand what happened), but usually would not cause harm.

The most dangerous operations - deletion and sending to spam, require two keypresses: shift-! or shift-3 (#).

But also dangerous: the `z` character is a single keypress: this is the undo shortcut and it can undo tagging (or untagging) operations without it being obvious to the user what happened. The `e` character (also a single keyress) is archiving, which strips all tags - not quite deletion, but close.

In short, keybindings can be surprising and the principle of least surprise dictates we should avoid such things. Should we disable keybindings unless people explicitly opt-in to them?

**Comment by Kabouik** [comment]
As long as they still can be enabled in the settings, I'm fine with it. But please don't fully remove them, I'm using some of them all the time and I would hate doing everything with the mouse. Maybe make the cheatsheet more obvious to users, I remember someone posting an issue here because he didn't know how to he had to press "?" to show it and instead was clicking on the symbol. If you add an option to toggle keybindings on and off, maybe there could be a clickable link to show the cheatsheet next to it (and keep the "?" keybinding too so that it can be displayed from anywhere in Mailpile).

Speaking of accidental manipulations, I am very glad that there is no key to directly send an email. Quite a few times I was composing an email and did shenanigans that brought me back to the inbox or any other place, but auto-save had my back most the time, and no emails have been sent accidentally. This is crucial. 

It would be great to actually have checkboxes next to each keyboard in the keyboard cheatsheet. There are many keybindings I will never use, and disabling them would limit the risk of accidents. However, I understand such a feature is very unlikely to happen.

**Comment by jamestomasino** [comment]
Seems like disabled by default--but providing a toggle to enable--is a easy and cheap safety mechanism. Those of us who value keyboard shortcuts are likely to go digging in settings to customize things anyway.

**Comment by BjarniRunar** [comment]
@Kabouik: don't worry, I use them constantly myself. Removing them couldn't be further from my mind. But making them opt-in is tempting, now that I've realized that they're an accident waiting to happen for some folks.

**Comment by JackDca** [comment]
I never use the key bindings. But also, I've never (as far as I know) had trouble caused by accidentally hitting one.
